$NetBSD: patch-darts.h.in,v 1.1 2026/06/06 05:15:17 adam Exp $

Fix for error: ISO C++17 does not allow 'register' storage class specifier.

--- darts.h.in.orig	2008-03-22 11:21:34.000000000 +0000
+++ darts.h.in
@@ -245,10 +245,10 @@ namespace Darts {
       T result;
       set_result(&result, -1, 0);
 
-      register array_type_  b = array_[node_pos].base;
-      register array_u_type_ p;
+      array_type_  b = array_[node_pos].base;
+      array_u_type_ p;
 
-      for (register size_t i = 0; i < len; ++i) {
+      for (size_t i = 0; i < len; ++i) {
         p = b +(node_u_type_)(key[i]) + 1;
         if (static_cast<array_u_type_>(b) == array_[p].check)
           b = array_[p].base;
@@ -272,12 +272,12 @@ namespace Darts {
                               size_t node_pos = 0) const {
       if (!len) len = length_func_()(key);
 
-      register array_type_  b   = array_[node_pos].base;
-      register size_t     num = 0;
-      register array_type_  n;
-      register array_u_type_ p;
+      array_type_  b   = array_[node_pos].base;
+      size_t     num = 0;
+      array_type_  n;
+      array_u_type_ p;
 
-      for (register size_t i = 0; i < len; ++i) {
+      for (size_t i = 0; i < len; ++i) {
         p = b;  // + 0;
         n = array_[p].base;
         if ((array_u_type_) b == array_[p].check && n < 0) {
@@ -310,8 +310,8 @@ namespace Darts {
                         size_t len = 0) const {
       if (!len) len = length_func_()(key);
 
-      register array_type_  b = array_[node_pos].base;
-      register array_u_type_ p;
+      array_type_  b = array_[node_pos].base;
+      array_u_type_ p;
 
       for (; key_pos < len; ++key_pos) {
         p = b + (node_u_type_)(key[key_pos]) + 1;
